### ISSUE TITLE
Form setValidation is not work correctly 

### ISSUE THREAD

**Issue by mbrostami** [phalcon/cphalcon#12465]
When I want to use setValidation for form, it doesn't validate elements!

CustomForm.php
```php
<?php

use Phalcon\Forms\Form;
use Phalcon\Forms\Element\Text;

class CustomForm extends Form
{
    public function initialize($entity = null, $options = array())
    {
        $this->add(new Text("name"));
    }
}
```
CustomValidation.php
```php
<?php

use Phalcon\Validation;
use Phalcon\Validation\Validator\PresenceOf;

class CustomValidation extends Validation
{
    public function initialize()
    {
        $this->add('name', new PresenceOf([
            'message' => 'Name is required'
        ]));
    }
}
```
Controller
```php
class IndexController extends Controller
{
    public function testAction()
    {
        $customForm = new CustomForm();
        $customForm->setValidation(new CustomValidation());
        $customForm->isValid($this->request->getPost()); // BUG: if you don't post name, this line returns true! 
    }
}
```

**Comment by Jurigag** [comment]
What a hell is `setValidation` method ?

**Comment by Jurigag** [comment]
https://github.com/phalcon/cphalcon/blob/master/phalcon/forms/form.zep#L256

This is current implementation of isValid method. It's just expecting any validators added to elements, if you didnt added anything then nothing will be validated.

**Comment by mbrostami** [comment]
So why there is a PUBLIC setter/getter for validation in form.zep!!!

**Comment by Jurigag** [comment]
I don't know.

**Comment by sergeyklay** [comment]
Fixed in the `3.0.x` branch. Thanks you
